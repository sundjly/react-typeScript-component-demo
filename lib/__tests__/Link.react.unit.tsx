import * as React from 'react';
import Link from '../React.link';
import * as renderer from 'react-test-renderer';

test('link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page='http://wwww.facebook.com'>facebook</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // @ts-ignore
  tree.props.onMouseEnter();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // @ts-ignore
  tree.props.onMouseLeave();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});