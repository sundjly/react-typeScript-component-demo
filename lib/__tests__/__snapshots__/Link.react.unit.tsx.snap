// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://wwww.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;

exports[`link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://wwww.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;

exports[`link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://wwww.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;
